File: ros2_ws/install/robovision_images/lib/robovision_images/my_subscriber.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from sensor_msgs.msg import Image
import cv2
from cv_bridge import CvBridge
import traceback


class ImageSubscriberNode(Node):
    def __init__(self):
        super().__init__("image_subscriber")

        self.img = []
        self.is_img = False
        self.display = True

        #Subscribers
        self.subscriber_ = self.create_subscription(
            Image, "camera/image", self.callback_camera_image, 10)
        
        #Processing
        self.processing_timer_ = self.create_timer(0.030, self.image_processing) #update image each 30 miliseconds

        self.get_logger().info("Starting image_subscriber application in python...")

    def callback_camera_image(self, msg):
        try:
            #Transform the new message to an OpenCV matrix
            bridge_rgb=CvBridge()
            self.img = bridge_rgb.imgmsg_to_cv2(msg,msg.encoding).copy()
            self.is_img = True

        except:
            self.get_logger().error(traceback.format_exc())

    def image_processing(self):
        if self.is_img:
            #Show your images
            if(self.display):
                cv2.imshow("view", self.img)
                cv2.waitKey(1)


def main(args=None):
    rclpy.init(args=args)
    node = ImageSubscriberNode()

    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        rclpy.shutdown()


if __name__ == "__main__":
    main()

File: ros2_ws/src/robovision_ros2/robovision_1/src/my_subscriber.cpp
#include "rclcpp/rclcpp.hpp"

#include <sensor_msgs/msg/image.hpp>
#include <cv_bridge/cv_bridge.h>
#include <opencv2/opencv.hpp>


class ImageSubscriberNode : public rclcpp::Node
{
public:
    ImageSubscriberNode() : Node("image_subscriber"), is_image_(false)
    {
        //Subscribers
        image_subscriber_ = this->create_subscription<sensor_msgs::msg::Image>(
            "/camera/rgb/image_raw", 10, std::bind(&ImageSubscriberNode::callback_image, this, std::placeholders::_1));
        
        //Processing
        image_processing_timer_ = this->create_wall_timer(
            std::chrono::milliseconds(30),
            std::bind(&ImageSubscriberNode::image_processing, this));

        RCLCPP_INFO(this->get_logger(), "Starting image_subscriber application in cpp...");
    }

private:
    
    bool is_image_;
    cv::Mat image_;
    sensor_msgs::msg::Image::SharedPtr image_msg_;

    rclcpp::Subscription<sensor_msgs::msg::Image>::SharedPtr image_subscriber_;
    rclcpp::TimerBase::SharedPtr image_processing_timer_;

    void callback_image(const sensor_msgs::msg::Image::SharedPtr msg)
    {
        image_ = cv_bridge::toCvCopy(msg, "bgr8")->image;
        is_image_ = true;
    }

    void image_processing()
    {
        if (is_image_){
            cv::imshow("view", image_);
            cv::waitKey(1);
        }
    }
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<ImageSubscriberNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
